src/Backend/Modules/Blog/Tests/Engine/ModelPostTest.php

```diff
@@ -0,0 +1,158 @@
+    }
+
+    public function testUpdatePost(): void
+    {
+        $postData = [
+            'id' => 1,
+            'revision_id' => Model::get(1)['revision_id'],
+            'meta_id' => Model::get(1)['meta_id'],
+            'category_id' => 1,
+            'user_id' => 1,
+            'language' => 'en',
+            'title' => 'title',
+            'introduction' => '<p>summary</p>',
+            'text' => '<p>main content</p>',
+            'publish_on' => '2017-11-23 09:23:00',
+            'created_on' => '2017-11-23 09:24:11',
+            'hidden' => 0,
+            'allow_comments' => true,
+            'num_comments' => 0,
+            'status' => 'active',
+            'edited_on' => '2017-11-23 09:24:11',
+        ];
+
+        $revisionId = Model::update($postData);
+        $updatedPost = Model::get(1);
+
+        $this->assertEquals($revisionId, $updatedPost['revision_id']);
+        $this->assertEquals($postData['id'], $updatedPost['id']);
+        $this->assertEquals($postData['category_id'], $updatedPost['category_id']);
+        $this->assertEquals($postData['user_id'], $updatedPost['user_id']);
+        $this->assertEquals($postData['meta_id'], $updatedPost['meta_id']);
+        $this->assertEquals($postData['language'], $updatedPost['language']);
+        $this->assertEquals($postData['title'], $updatedPost['title']);
+        $this->assertEquals($postData['introduction'], $updatedPost['introduction']);
+        $this->assertEquals($postData['text'], $updatedPost['text']);
+        $this->assertArrayHasKey('image', $updatedPost);
+        $this->assertEquals($postData['status'], $updatedPost['status']);
+        $this->assertArrayHasKey('publish_on', $updatedPost);
+        $this->assertArrayHasKey('created_on', $updatedPost);
+        $this->assertEquals(strtotime($postData['edited_on']), $updatedPost['edited_on']);
+        $this->assertEquals($postData['hidden'], $updatedPost['hidden']);
+        $this->assertEquals('1', $updatedPost['allow_comments']);
+        $this->assertEquals($postData['num_comments'], $updatedPost['num_comments']);
+        $this->assertArrayHasKey('url', $updatedPost);
+    }
+
+    public function testUpdatePostRevision(): void
+    {
+        Model::updateRevision(
+            Model::get(1)['revision_id'],
+            [
+                'image' => 'foo.jpg',
+            ]
+        );
+
+        $updatedRevision = Model::get(1);
+        $this->assertEquals($updatedRevision['image'], 'foo.jpg');
+    }
+
+    public function testDeletePost(): void
+    {
+        $this->assertTrue(Model::exists(1));
+        Model::delete([1]);
+        $this->assertFalse(Model::exists(1));
+    }
+
+    public function testCalculatingPostUrl(): void
+    {
+        $this->assertEquals('foo-bar', Model::getUrl('foo-bar'));
+
+        // check if 2 is appended for an existing post
+        $this->assertEquals('blogpost-for-functional-tests-2', Model::getUrl('blogpost-for-functional-tests'));
```

Comment:
this blog post doesn't exist anymore, as you deleted it in "testDeletePost"?

Reply:
No, fixtures are reloaded on each test by the setUp method